homeassistant/components/meteo_france/config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Config flow to configure the Meteo-France integration."""
+import logging
+
+from meteofrance.client import meteofranceClient, meteofranceError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+
+from .const import CONF_CITY, SENSOR_TYPES
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MeteoFranceFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Meteo-France config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Meteo-France config flow."""
+
+    def _configuration_exists(self, city_name: str) -> bool:
+        """Return True if city_name exists in configuration."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_CITY] == city_name:
+                return True
+        return False
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_CITY, default=user_input.get(CONF_CITY, "")): str}
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            return await self._show_setup_form(user_input, errors)
+
+        city = user_input[CONF_CITY]
+        monitored_conditions = user_input.get(
+            CONF_MONITORED_CONDITIONS, list(SENSOR_TYPES)
+        )
```

Comment:
Create properties for these local variables.

Reply:
What to you mean ?

I'm using those vars only in this function.

Reply:
For example you use city again in line 77 and for consistency I also wanted to extract monitored_conditions as well. Therefore, you could declare them as attributes in the constructor of this class.

Reply:
Understood, but I don't agree.

There is integration without properties (and a lot) : 
![Capture d’écran 2020-01-11 à 20 12 49](https://user-images.githubusercontent.com/14821482/72209494-15547880-34af-11ea-8c92-7f2925c5d0d9.png)


I am more willing to remove `monitored_conditions` here and use `user_input.get(CONF_MONITORED_CONDITIONS, list(SENSOR_TYPES))` directly inside the data parameter of `async_create_entry`, because it's used only here so no need to keep that information in memory.

Reply:
Again I have to agree with you, it's fine as it is 😄 sorry my bad.